text/0000-advance-security-disclosure.md

```diff
@@ -0,0 +1,317 @@
+external to the organization without explicit permission from the WG.
+
+Organizations must take appropriate steps to protect the confidentiality of
+advance notifications and of all communications and materials associated with
+them. If the organization already has established procedures for dealing with
+confidential information received from partners, those procedures should at
+least apply (in addition to any other procedure deemed necessary).
+
+There will be no formal NDA to sign to start receiving early notifications, but
+any violation of confidentiality will still have consequences (see "Removal
+from the list").
+
+## Public pre-announcement of security releases
+
+If a security release is planned to address the vulnerability, at the same time
+of advance disclosure the Security Response WG will post a pre-announcement of
+the upcoming security release on the [public security announces mailing
+list][announce-ml], the [Rust blog][blog] and any other venue the WG deems
+appropriate. The announcement should mention:
+
+* The planned date and time (with the time zone) for the release.
+* The severity of the vulnerability, as determined by the WG.
+* The affected part of the project (e.g., the compiler, the standard library...).
+
+The Security Response WG can opt to mention less information than that, if it
+deems sharing more would increase the likelihood of the vulnerability being
+independently rediscovered too much.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The Rust Security Response WG is currently well respected in the industry, and
+we are trusted with reports about cross-cutting vulnerabilities affecting not
+just Rust. This respect and trust is fragile, and any leak caused (indirectly)
+by the WG could reduce the amount of vulnerabilities shared with us, worsening
+the security of the ecosystem as a whole.
+
+The risk of leaks will increase as we notify more organizations ahead of time.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The pillar that guided the drafting of this RFC is *"make sure every Rust
+developer has access to an updated toolchain when the vulnerability is
+disclosed"*. This gives all players in the ecosystem the same information
+required to address and fix the vulnerabilities, and in the WG's opinion this
+is the right balance between sharing with the least amount of people and having
+the biggest impact.
+
+The WG considered multiple alternatives, listed below.
+
+## Keep the status quo
+
+Do nothing, and continue with the current policy of only sharing
+vulnerabilities ahead of time with [distros@vs.openwall.org][distros]. This
+doesn't cover non-Linux distributions of Rust.
+
+## Don't share with anyone
+
+Restrict our policies even more and don't share vulnerabilities ahead of time
+with anyone. This would put every Rust developer who doesn't use the toolchain
+provided by the project itself at risk whenever a vulnerability is announced.
+
```

Comment:
```suggestion

The use of toolchains not provided directly by the Rust project has use cases that are critical for Rust's adoption in ways we care about. For instance, the use of Rust in place of other languages in operating systems, system-level tooling, and C libraries depends on the ability to build such code using toolchains provided by the same source (in order for distributions to remain self-contained), and not supporting such redistribution would hamper adoption of Rust in these use cases.

```

Reply:
Added the mention about system-level tooling in distributions, plus a mention of another sector that the Rust-provided toolchains wouldn't be able to cater (qualified industries requiring 10 to 20 years of long-term support for a toolchain).